tests/test_security.py

```
@@ -0,0 +1,98 @@
+        with tempfile.TemporaryDirectory() as tmpdir:
+            # Create a malicious zip with path traversal
+            zip_path = os.path.join(tmpdir, "malicious.zip")
+            extract_to = os.path.join(tmpdir, "extract")
+            os.makedirs(extract_to)
+
+            with zipfile.ZipFile(zip_path, 'w') as zf:
+                # Try to write outside the extraction directory
+                zf.writestr("../../evil.txt", "malicious content")
+
+            # This should raise ValueError due to path traversal
+            with pytest.raises(ValueError, match="path traversal|Invalid path"):
```

Ruff (0.14.10)

Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)